File: Lab_7/Codes/Open_Closed.py
import cv2
import numpy as np
import matplotlib.pyplot as plt

# Load the image
original_image_rgb = cv2.imread('baby2.jpg')
# Convert the image from BGR (OpenCV format) to RGB (matplotlib format)
original_image_rgb = cv2.cvtColor(original_image_rgb, cv2.COLOR_BGR2RGB)

# Define the structuring element (disk-shaped)
se = cv2.getStructuringElement(cv2.MORPH_ELLIPSE, (5, 5))

# Perform Opening and Closing on each channel
opened_image = np.zeros_like(original_image_rgb)
closed_image = np.zeros_like(original_image_rgb)
for channel in range(3):
    opened_image[:, :, channel] = cv2.morphologyEx(original_image_rgb[:, :, channel], cv2.MORPH_OPEN, se)
    closed_image[:, :, channel] = cv2.morphologyEx(original_image_rgb[:, :, channel], cv2.MORPH_CLOSE, se)

# Display the images
plt.figure(figsize=(15, 5))
plt.subplot(1, 3, 1)
plt.imshow(original_image_rgb)
plt.title('Original Image')
plt.subplot(1, 3, 2)
plt.imshow(opened_image)
plt.title('Opened Image')
plt.subplot(1, 3, 3)
plt.imshow(closed_image)
plt.title('Closed Image')
plt.show()

File: Lab_7/Codes/Open_Closed.m
% Read the original image
originalImage = imread('baby2.jpg');
% Define a structuring element
se = strel('disk', 5); % You can adjust the size of the disk
% Initialize output images
openedImage = zeros(size(originalImage), 'like', originalImage);
closedImage = zeros(size(originalImage), 'like', originalImage);
% Perform Opening and Closing on each channel
for channel = 1:size(originalImage, 3)
    openedImage(:, :, channel) = imopen(originalImage(:, :, channel), se);
    closedImage(:, :, channel) = imclose(originalImage(:, :, channel), se);
end
% Create a figure to display the images
figure;
% Display the original image
subplot(1, 3, 1);
imshow(originalImage);
title('Original Image');
% Display the opened image
subplot(1, 3, 2);
imshow(openedImage);
title('Opened Image');
% Display the closed image
subplot(1, 3, 3);
imshow(closedImage);
title('Closed Image');